File: Seminar4/python/student_app/domain/student.py
from domain.person import Person


class Student(Person):#Comparable
    generalId = 0


    def __init__(self, name, age):
        super().__init__(name, age)
        self.id = Student.generalId
        Student.generalId += 1

    def getId(self):
        return self.id

    def __str__(self):
        return "Students [age=" + str(super().getAge()) + ", name=" + super().getName() + ", id=" + str(self.id) + "]"

    def __lt__(self, obj): 
        return ((self.age) < (obj.age)) 
  
    def __gt__(self, obj): 
        return ((self.age) > (obj.age)) 
  
    def __le__(self, obj): 
        return ((self.age) <= (obj.age)) 
  
    def __ge__(self, obj): 
        return ((self.age) >= (obj.age)) 
  
    def __eq__(self, obj): 
        return (self.age == obj.age) 

File: Seminar6/GenericInherence/GIApp/src/Student.java


//T - string
//V - int
public class Student<T extends Comparable<T>,V> extends Person<T,V> implements Comparable<Student<T,V>> {

    private V id;
   // private static int generalId;

    public Student(T name, V age, V id) {
        super(name, age);
        this.id = id;
        //this.id = generalId;
       // generalId++;
    }


    public V getId() {
        return id;
    }


    @Override
    public String toString() {
        return "Students [age=" + super.getAge() + ", name=" + super.getName() + ", id=" + id + "]";
    }

    @Override
    public int compareTo(Student<T,V> o) {

        return super.getName().compareTo(o.getName());

        // System.out.println(super.getName()+" - "+o.getName());
        // if(super.getAge()==o.getAge())
        // {
        //     id.
        //     if(id==o.id)return 0 ;
        //     if(id>o.id)return 1;
        //     else return -1;
        //     //return 0;
        // }

        // if(super.getAge()>o.getAge())
        // return 1;
        // else
        // return -1;        
    }


    
    
    
}


